### PR TITLE
entc/gen: add option to extend existing templates

### PR DESCRIPTION
**Pull request by a8m**

### CODE REVIEW

File: entc/integration/ent/template/meta.tmpl [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+{{ define "meta/additional/enums" }}
+	{{- range $f := $.Fields }}
+		{{- if $f.IsEnum }}
+			{{ $enum := trimPackage $f.Type.String $.Package }}
+			{{ $receiver := receiver $f.BuilderField }}
+
+			// Ptr returns a new pointer to the enum value.
+			func ({{ $receiver }} {{ $enum }}) Ptr() *{{ $enum }} {
+				v := {{ $receiver }}
+				return &v
```
Comment (alexsn) [inline entc/integration/ent/template/meta.tmpl:11]:
```suggestion
				return &{{ $receiver }}
```

File: entc/integration/ent/template/meta.tmpl [hunk 2]
```diff
@@ -0,0 +1,20 @@
+{{/* A template for adding additional fields to specific types. */}}
+{{ define "meta/additional/enums" }}
+	{{- range $f := $.Fields }}
+		{{- if $f.IsEnum }}
+			{{ $enum := trimPackage $f.Type.String $.Package }}
```
Comment (alexsn) [inline entc/integration/ent/template/meta.tmpl:5]:
```suggestion
			{{ $enum := trimPackage $f.Type.String $.Package -}}
```

File: entc/integration/ent/template/meta.tmpl [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+			{{ $receiver := receiver $f.BuilderField }}
```
Comment (alexsn) [inline entc/integration/ent/template/meta.tmpl:6]:
```suggestion
			{{ $receiver := receiver $f.BuilderField -}}
```

File: entc/integration/ent/template/meta.tmpl (hunk 2 above)
Comment (a8m) [reply entc/integration/ent/template/meta.tmpl:5]: It's not really needed here, because gofmt will remove the spaces between methods.

**Pull request merged**